_ml-commons-plugin/conversational-search.md

```diff
@@ -0,0 +1,209 @@
+
+```jsx
+.plugins-ml-conversation-interactions
+{
+    "_meta": {
+        "schema_version": 1
+    },
+    "properties": {
+        "conversation_id": {"type": "keyword"},
+        "create_time": {"type": "date", "format": "strict_date_time||epoch_millis"},
+        "input": {"type": "text"},
+        "prompt_template": {"type": "text"},
+        "response": {"type": "text"},
+        "origin": {"type": "keyword"},
+        "additional_info": {"type": "text"}
+    }
+}
+```
+
+### Creating conversations and interactions
+
+To begin using conversation memory, enable the following cluster setting:
+
+```json
+PUT /_cluster/settings
+{
+  "persistent": {
+    "plugins.ml_commons.memory_feature_enabled": true
+  }
+}
+```
+
+After conversation memory is enabled, you can now use the Memory API to create a conversation. Remember, you cannot create an interactions until you've created a conversation. 
+
+To make the conversation easily identifable, use the optional `name` field in the Memory API, as shown in the following example. This will be your only opportunity to give your conversation a name.
+
+
+
+```json
+POST /_plugins/_ml/memory/conversation
+{
+  "name": Example conversation
+}
+```
+
+The Memory API responds with the conversation ID, as shown in the following example response:
+
+```json
+{ "conversation_id": "4of2c9nhoIuhcr" }
+```
+
+You'll use the `conversation_id` to create interactions inside the conversation. To create interactions, enter the `conversation_id` into the Memory API path. Then, customize the [fields](#conversation-interactions-index) in the request body, as shown in the following example:
+
+```json
+POST /_plugins/_ml/memory/conversation/4of2c9nhoIuhcr
+{
+  "input": "How do I make an interaction?",
+  "prompt_template": "Hello OpenAI, can you answer this question? \
+											Here's some extra info that may help. \
+											[INFO] \n [QUESTION]",
+  "response": "Hello, this is OpenAI. Here is the answer to your question.",
+  "origin": "MyFirstOpenAIWrapper",
+  "additional_info": "Additional text related to the answer \
+											A JSON or other semi-structured response"
+}
+```
+
+The Memory API then responds with an interaction ID, as shown in the following response:
+
+```json
+{ "interaction_id": "948vh_PoiY2hrnpo" }
+```
+
+### Getting conversations
+
+You can get a list of conversation using the following Memory API operation:
+
+```json
+GET /_plugins/_ml/memory/conversation?max_results=3&next_token=0
+```
+
+Use the following path parameters to customize your results:
+
+Parameter | Data type | Description
+:--- | :--- | :---
+`max_results` | Integer | The maximum number of results returned by the response. Default is `10`.
+`next_token` | Integer | The number of paginated results returned when more than the `max_results` exist.
```

Comment:
`next_token` represents the position in the conversation ordering to retrieve. It functions like the `from` field in the index `_search` endpoint. So if my conversations are
A, B, C, D, E,
Then specifying `next_token=2` would return conversations starting at C, e.g.
C, D, E

`next_toke` does not represent the number of paginated results.